xemutest/test_nxdk_pgraph_tests.py

```diff
@@ -0,0 +1,71 @@
+"""Test harness for nxdk_pgraph_tests."""
+
+import logging
+import os
+import shutil
+
+from pyfatx import Fatx
+import test_base
+
+log = logging.getLogger(__file__)
+
+class TestNXDKPgraphTests(test_base.TestBase):
+    """Runs the nxdk_pgraph_tests suite and validates output."""
+
+    def __init__(self, test_env: test_base.TestEnvironment, results_path: str, test_data_path: str):
+        shutil.rmtree(results_path)
+        self.config_file_path = os.path.join(test_data_path, 'config.cnf')
+        if not self.config_file_path:
+            raise FileNotFoundError(
+                'Test data was not installed with the package. You need to copy '
+                f'config.cnf to {test_data_path}.')
+
+        iso_path = os.path.join(test_data_path, 'nxdk_pgraph_tests.iso')
+        if not os.path.isfile(iso_path):
+            raise FileNotFoundError('Test data was not installed with the package. You need to build it and copy '
+                                    f'to {test_data_path}.')
+
+        self.golden_results_path = os.path.join(test_data_path, "golden_results")
+
+        timeout_seconds = 20 * 60
```

Comment:
Do the tests really take 20 minutes?

Reply:
I tested and it completes within a minute or so. How about we lower this to like 5 minutes?

Reply:
It runs in 5 minutes because I've only whitelisted one test for demo purposes. I suspect if we enable all the tests it'll actually take more than 20 minutes (though we can probably be selective about the subset of tests and the tester is probably overly conservative in waiting when saving the framebuffer). I think we may actually need to do something more creative wrt. sharding the tests and running multiple times.

Happy to set it to 5 mins for now, but I'm confident it'll need to be bumped up as tests are turned on.

Reply:
After some optimization work, I can run all the tests in your suite in just under 5 minutes on my development machine (<90s if not running the depth format tests). The VPS which is used for testing is considerably slow, so we will need to see what the average runtime for it is. Hopefully not 20 minutes though.

Reply:
Just saw your PR, awesome improvement :)